# Fix for the NSLU2 timer problem (tickadj 10101.01 when it should be 10000)
# Also contains a fix for the more generate problem on the ixp4xx in that the
# Intel spec says to use a 33.33MHz crystal, not 33.333333Mhz

--- linux-2.6.12.2/include/asm-arm/arch-ixp4xx/timex.h.orig	2005-08-05 11:02:09.872690723 -0700
+++ linux-2.6.12.2/include/asm-arm/arch-ixp4xx/timex.h	2005-08-05 11:45:08.786971567 -0700
@@ -7,7 +7,14 @@
 
 /*
  * We use IXP425 General purpose timer for our timer needs, it runs at 
- * 66.66... MHz
+ * 66.66 MHz (exactly) by the Intel specification.
  */
-#define CLOCK_TICK_RATE (66666666)
-
+#if 0
+#define CLOCK_TICK_RATE (66660000)
+#else
+/*
+ * Unfortunately the NSLU2 has a 33MHz crystal, so the rate is not the
+ * specified one.
+ */
+#define CLOCK_TICK_RATE (66000000)
+#endif
